### PR TITLE
feat(deps): update to backstage 1.29

### PR DESCRIPTION
**Pull request by schultzp2020**
## Description

Please explain the changes you made here.

## Which issue(s) does this PR fix

- Fixes https://issues.redhat.com/browse/RHIDP-2791

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: packages/backend/src/index.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import { rootHttpRouterServiceFactory } from '@backstage/backend-app-api';
 import { statusCheckHandler } from '@backstage/backend-common';
```
Comment (schultzp2020) [inline packages/backend/src/index.ts:1]: @davidfestal @gashcrumb can you please help me fix these deprecations?

Comment (gashcrumb) [reply packages/backend/src/index.ts:1]: Yep, this line looks like the export has been moved and should now be:

```
import { rootHttpRouterServiceFactory } from '@backstage/backend-defaults/rootHttpRouter'
```

And probably we have to be careful what `@backstage/backend-defaults` version is pulled in from this point on.

Comment (schultzp2020) [reply packages/backend/src/index.ts:1]: @gashcrumb what about `import { statusCheckHandler } from '@backstage/backend-common';`?

Comment (gashcrumb) [reply packages/backend/src/index.ts:1]: Hmm, it looks like `statusCheckHandler` is going to be removed in favor of the ["Root Health Service"](https://backstage.io/docs/backend-system/core-services/root-health/) which I think we will need to this code to use instead.

File: dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-bitbucket-cloud/dist-dynamic/package.json [hunk 2]
```diff
@@ -12,7 +12,9 @@
   },
   "backstage": {
     "role": "backend-plugin-module",
-    "supported-versions": "1.26.5"
+    "supported-versions": "1.27.7",
```
Comment (nickboldt) [inline dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-bitbucket-cloud/dist-dynamic/package.json:15]: this should move to 1.29.latest, right? If you prefer I can do that with a sed PR after this is merged.

**Comment by nickboldt** [comment]
all tests passed except prow, so triggered them again after latest update.

**Comment by schultzp2020** [comment]
Received approval from QE to merge with e2e tests failing.

**Pull request merged**